src/librustdoc/json/conversions.rs

```diff
@@ -760,20 +760,24 @@ impl FromWithTcx<clean::Discriminant> for Discriminant {
 
 impl FromWithTcx<clean::Import> for Import {
     fn from_tcx(import: clean::Import, tcx: TyCtxt<'_>) -> Self {
-        use clean::ImportKind::*;
-        let (name, glob) = match import.kind {
-            Simple(s) => (s.to_string(), false),
-            Glob => (
+        let (name, glob, id) = match import.kind {
+            ImportKind::Simple(s) => {
+                let is_from_doc_hidden =
+                    import.imported_item_is_doc_hidden(tcx) || import.inherits_doc_hidden(tcx);
```

Comment:
NIT: The definition of `inherits_doc_hidden` already calls `imported_item_is_doc_hidden`. That should be de-duplicated, either here or there.